app/controllers/declaration/AdditionalInformationRequiredController.scala

```diff
@@ -44,24 +47,32 @@ class AdditionalInformationRequiredController @Inject()(
     Ok(additionalInfoReq(mode, itemId, YesNoAnswer.form()))
   }
 
-  def submitForm(mode: Mode, itemId: String): Action[AnyContent] = (authenticate andThen journeyType) { implicit request =>
+  def submitForm(mode: Mode, itemId: String): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
     YesNoAnswer
       .form()
       .bindFromRequest()
       .fold(
-        (formWithErrors: Form[YesNoAnswer]) => BadRequest(additionalInfoReq(mode, itemId, formWithErrors)),
-        validYesNo => navigator.continueTo(mode, nextPage(yesNoAnswer = validYesNo, itemId))
+        (formWithErrors: Form[YesNoAnswer]) => Future.successful(BadRequest(additionalInfoReq(mode, itemId, formWithErrors))),
+        validYesNo =>
+          updateCache(validYesNo, itemId).map { _ =>
+            navigator.continueTo(mode, nextPage(validYesNo, itemId))
+        }
       )
-
+  }
+  private def updateCache(yesNoAnswer: YesNoAnswer, itemId: String)(implicit r: JourneyRequest[AnyContent]): Future[Option[ExportsDeclaration]] = {
+    val updatedAdditionalInformation = yesNoAnswer.answer match {
+      case YesNoAnswers.yes => r.cacheModel.itemBy(itemId).flatMap(_.additionalInformation).getOrElse(AdditionalInformationData(Seq.empty))
```

Comment:
If "Yes" then we either keep the existing data or create a new AdditionalInformationData(Seq.empty)
If "No" we save a new AdditionalInformationData(Seq.empty) which will delete previous data if user changed from Yes to No